src/main.cpp

```diff
@@ -29,32 +29,40 @@ int main(int argc, char *argv[])
 	Initialize(window);
 
 	Camera* camera = new Camera(glm::vec3(4,3,3), glm::vec3(0,0,0), glm::vec3(0,1,0));
-	InputManager inputManager(window, camera);
 
 	// Create Geometry
 	Geometry* geometry = new Geometry("models/cube/cube.obj", GL_TRIANGLES);
 
 	// Create Camera
-	// Create Scene
-	Scene scene("shaders/vertex.glsl", "shaders/fragment.glsl", camera);
-	scene.AddGeometry(geometry);
+	Shader program("shaders/vertex.glsl","shaders/fragment.glsl");
+	Shader programOutline("shaders/vertex.glsl","shaders/outline.frag");
+	Scene* scene = new Scene(&program, &programOutline, camera);
+	scene->AddGeometry(geometry);
+
+	
+	InputManager inputManager(window, camera, scene);
 
 	CheckGLErrors();
 
+	glEnable(GL_DEPTH_TEST);
+	glEnable(GL_STENCIL_TEST);
+	glStencilFunc(GL_NOTEQUAL, 1, 0xFF);
+	glStencilOp(GL_KEEP, GL_KEEP, GL_REPLACE);
```

Comment:
These are better off in GLHelpers::Initialize instead of main.

Also, are the enables here even necessary? Doesn't these get called by Render and/or RenderStencil in each frame anyways?

Reply:
d

Reply:
moved to GLHelpers::Initialize

removed 
+	glEnable(GL_DEPTH_TEST);
+	glEnable(GL_STENCIL_TEST);

other 2 is necessary